### PR TITLE
Participation and Communication

### PR DESCRIPTION
**Pull request by ullrichd21**
## Description

This PR adds the participation and communication sections of the code of conduct.

### Type of Change

- [ ] Feature
- [ ] Bug fix
- [x] Documentation

## Contributors

- @ullrichd21
- @pkelly55
- @liamblack3
- @jackturner83

## Reminder

All GitHub Actions checks must be in a passing state in this branch or fork before this pull request can be merged.

### CODE REVIEW

File: conduct.md [hunk 1]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
 
-TODO: Write an introductory paragraph.
+Communication is key to both successful and sustainable code. However it's important that all communication between students and team members follows the three pillars of Respect, Trust, and Humility. In general, students will *respect* their peers, *trust* their contributions and thoughts, and practice *humility* in all aspects of communication, in person, on GitHub, in the Discord, and otherwise.
 
 #### Acceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Team members communitcating in a professional and respectful manner.
```
Comment (nathandloria) [inline conduct.md]: Please remove space after the period and fix the spelling of communicating.

File: conduct.md [hunk 2]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
+Communication is key to both successful and sustainable code. However it's important that all communication between students and team members follows the three pillars of Respect, Trust, and Humility. In general, students will *respect* their peers, *trust* their contributions and thoughts, and practice *humility* in all aspects of communication, in person, on GitHub, in the Discord, and otherwise.
 
 #### Acceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Team members communitcating in a professional and respectful manner. 
+- Team members should use words like "Please" and "Thank you".
+- Team members will allow other members and teams speak freely.
```
Comment (nathandloria) [inline conduct.md]: Please remove the space after the period.

File: conduct.md [hunk 3]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
+- Team members communitcating in a professional and respectful manner. 
+- Team members should use words like "Please" and "Thank you".
+- Team members will allow other members and teams speak freely. 
 
 #### Unacceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Team members communicate in a way which offends or disrespects others.
```
Comment (nathandloria) [inline conduct.md]: Please remove the space after the period.

File: conduct.md [hunk 4]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
+- Team members should use words like "Please" and "Thank you".
+- Team members will allow other members and teams speak freely. 
 
 #### Unacceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Team members communicate in a way which offends or disrespects others. 
+- Team members constantly interrupting others - we all have something to share.
```
Comment (nathandloria) [inline conduct.md]: Please remove the space after the period.

File: conduct.md [hunk 5]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
+- Team members will allow other members and teams speak freely. 
 
 #### Unacceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Team members communicate in a way which offends or disrespects others. 
+- Team members constantly interrupting others - we all have something to share. 
+- Team members show signs of passive aggresive behaivor. Working as a team means keeping emotions out of work.
```
Comment (nathandloria) [inline conduct.md]: Please fix the spelling of the word aggressive and behavior.

File: conduct.md (hunk 2 above)
Comment (noorbuchi) [inline conduct.md]: I think that rephrasing acceptable conduct to have the verb at the beginning of each bullet point sounds better and can be standardized. However, I'll leave that up to your group and the class in general to decide.

Currently, the structure of the statements is not consistent in verb tenses.

For example here is how one of these bullet points can be restructured (From the section below):
Instead of:
```
Team members communicate in a way which offends or disrespects others. 
```
could be rephrased to:
```
Communicating in a way which offends or disrespects others. 
```
In most of these cases it's assumed that the conduct applies to everyone in the class.

**Review by gvanzin-allegheny — APPROVED** [review]
Looks good.

File: conduct.md [hunk 6]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
 
 ### Participation
 
-TODO: Write an introductory paragraph.
+Participation for students will be a multidide of factors. We all have bad days and  we all have topics we know more or less about but how many times a student speaks a day will not soley make up their participation. It is being prepared each day, having done the homework, coming to class with an open mind to learn and making an effort to be apart of either small group or large group conversations that will reflect students participation.
```
Comment (nathandloria) [inline conduct.md]: Please fix the spelling of multitude and solely.

File: conduct.md [hunk 7]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
 
 #### Acceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Students will respect all peers opinions.
```
Comment (noorbuchi) [inline conduct.md]: In the previous category `Team members` was used. However, `Students` is used here. Is there a difference between the two, or are they referring to the same people. I think it's important to clarify that.

File: conduct.md [hunk 8]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
+- Students participation will be reflected in participation surveys.
 
 #### Unacceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Students will not interrupt their peers.
+- Students will not skip reading or assignments in general.
+- Students will not disrepect peer contributions in work, code, speech, or otherwise.
```
Comment (nathandloria) [inline conduct.md]: Please fix the spelling of the word disrespect.

**Review by nathandloria — CHANGES_REQUESTED** [review]
Please review these changes.

**Review by gvanzin-allegheny — DISMISSED** [review]
There are some spelling issues in the sentences.

**Review by noorbuchi — CHANGES_REQUESTED** [review]
Thank you for your work on this PR. Let's discuss some of the comments I brought up to agree on a standard to use before merging. 

Additionally, can you please fix the errors in the automated checks. Thanks!

**Review by nathandloria — DISMISSED** [review]
I think this Pull Request looks good! The changes that you made are satisfactory for me.

File: conduct.md [hunk 9]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
 
 ### Communication
 
-TODO: Write an introductory paragraph.
+Communication is key to both successful and sustainable code. However it's important that all communication between students and team members follows the three pillars of Respect, Trust, and Humility. In general, students will *respect* their peers, *trust* their contributions and thoughts, and practice *humility* in all aspects of communication, in person, on GitHub, in the Discord, and otherwise.
```
Comment (solisa986) [inline conduct.md]: it looks like students and team members are used interchangeably. If there is a difference between the two, there should be a sentence clarifying the distinction.

File: conduct.md [hunk 10]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
+- Students should use words like "Please" and "Thank you".
+- Students will allow other members and teams speak freely.
 
 #### Unacceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Students should not communicate in a way that offends or disrespects others.
+- Students constantly interrupting others - we all have something to share.
```
Comment (solisa986) [inline conduct.md]: This comment should follow the same syntax as the other ones. Ergo, "Students should", "Students will", etc.

**Review by nathandloria — DISMISSED** [review]
As long as Noor's suggestions have been resolved, I am fine with this being merged!

File: conduct.md (hunk 9 above)
Comment (noorbuchi) [inline conduct.md]: I think this line is a bit long. It doesn't look like it's triggering a markdown lint error but it would be nice to break it up on multiple lines.

File: conduct.md [hunk 11]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
-- TODO
-- TODO
+- Students should not communicate in a way that offends or disrespects others.
+- Students should not interrupt others - we all have something to share.
+- Students should not show signs of passive aggressive behavior. Working as a team means keeping emotions out of work.
 
 ### Participation
 
-TODO: Write an introductory paragraph.
+Participation for students will be a multitude of factors. We all have bad days and  we all have topics we know more or less about but how many times a student speaks a day will not solely make up their participation. It is being prepared each day, having done the homework, coming to class with an open mind to learn and making an effort to be apart of either small group or large group conversations that will reflect students participation.
```
Comment (noorbuchi) [inline conduct.md]: same thing with line length

**Review by noorbuchi — DISMISSED** [review]
Aside from the line length comments, I think the PR looks great!
Thanks for your contribution to the code of conduct.

**Review by solisa986 — CHANGES_REQUESTED** [review]
once these changes are done, the pr looks good!

File: conduct.md (hunk 9 above)
Comment (ullrichd21) [reply conduct.md]: I personally don't think it needs to be broken up more, I think it's fine as one thought.

File: conduct.md (hunk 11 above)
Comment (ullrichd21) [reply conduct.md]: I don't think it's an issue since it's not a run on so I'm going to mark this suggestion resolved.

File: conduct.md (hunk 9 above)
Comment (ullrichd21) [reply conduct.md]: I mentioned team members to drive in that it's expected of all students and also in the group setting, therefor, as long as it isn't confusing, I'm going to keep this line how it is.

File: conduct.md (hunk 10 above)
Comment (ullrichd21) [reply conduct.md]: This line is outdated and has since been changed! 👍🏻

**Review by PaigeCD — DISMISSED** [review]
I think this is fine.

File: conduct.md (hunk 9 above)
Comment (mariakimheinert) [inline conduct.md]: A comma should be added after "However".

File: conduct.md [hunk 12]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
+- Students will allow other members and teams speak freely.
 
 #### Unacceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Students should not communicate in a way that offends or disrespects others.
+- Students should not interrupt others - we all have something to share.
+- Students should not show signs of passive aggressive behavior. Working as a team means keeping emotions out of work.
```
Comment (mariakimheinert) [inline conduct.md]: I'm not sure about the last part, "Working as a team means keeping emotions out of work." It might promote tone policing. One of our textbooks, [Building Software Together](https://buildtogether.tech/important/), includes a short section on tone policing.

> **Tone policing** is a tactic used to shut down discussion by criticizing people for expressing emotion. It is often used against women, who are told to "calm down" when they show any sign of being angry at being excluded or not taken seriously. As [this article](https://thebias.com/2017/09/26/how-good-intent-undermines-diversity-and-inclusion/) by [Annalee Flower](https://flowerhorne.com/) says:

> People often reach for positive statements like "assume good intent" because they're worried about people being "shamed" over innocent mistakes. But society at large is already inclined to assume good intent in people with power and privilege–even when they're not demonstrating it. If you want to build a culture of "assuming good intent," start by assuming good intent in marginalized people.

> Assume that they already tried being nice. Assume that their feelings are valid. Assume that, after a lifetime of practice, they are responding to harmful behavior in the way that is safest for them. Prioritize that safety over the momentary discomfort people feel when they realize they've done something hurtful.

Thoughts?

File: conduct.md [hunk 13]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
+Participation for students will be a multitude of factors. We all have bad days and  we all have topics we know more or less about but how many times a student speaks a day will not solely make up their participation. It is being prepared each day, having done the homework, coming to class with an open mind to learn and making an effort to be apart of either small group or large group conversations that will reflect students participation.
 
 #### Acceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Students will respect all peers opinions.
+- Students will show up to every class ready to learn and engage with the class.
+- Students participation will be reflected in participation surveys.
```
Comment (mariakimheinert) [inline conduct.md]: Could this sentence be reworded to be an action that a student takes? Perhaps a sentence about how students will evaluate their own participation and the participation of their group members by completing the participation survey at the end of each week?

File: conduct.md (hunk 11 above)
Comment (mariakimheinert) [inline conduct.md]: There is an extra space after "We all have bad days and". Also, it would be best to add a comma after "we all have topics we know more or less about".

Comment (mariakimheinert) [reply conduct.md]: Instead of "homework", can we use a phrase that more precisely reflects the type of work that is assigned in this course?

File: conduct.md [hunk 14]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
 
-TODO: Write an introductory paragraph.
+Participation for students will be a multitude of factors. We all have bad days and  we all have topics we know more or less about but how many times a student speaks a day will not solely make up their participation. It is being prepared each day, having done the homework, coming to class with an open mind to learn and making an effort to be apart of either small group or large group conversations that will reflect students participation.
 
 #### Acceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Students will respect all peers opinions.
```
Comment (mariakimheinert) [inline conduct.md]: There should be an apostrophe after "peers".

File: conduct.md [hunk 15]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
+- Students will show up to every class ready to learn and engage with the class.
+- Students participation will be reflected in participation surveys.
 
 #### Unacceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Students will not interrupt their peers.
+- Students will not skip reading or assignments in general.
```
Comment (mariakimheinert) [inline conduct.md]: The "in general" seems to suggest that it's okay the skip reading *some* of the time.
Since reading assignments are most critical to being able to participate effectively in class, could this be reworded to state that students should not miss reading assignments?

File: conduct.md [hunk 16]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
+- Students should use words like "Please" and "Thank you".
+- Students will allow other members and teams speak freely.
 
 #### Unacceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Students should not communicate in a way that offends or disrespects others.
+- Students should not interrupt others - we all have something to share.
```
Comment (mariakimheinert) [inline conduct.md]: The dash should be two dashes that do not have spaces around them: "others - we" to "others--we".

File: conduct.md [hunk 17]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
+- Students participation will be reflected in participation surveys.
 
 #### Unacceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Students will not interrupt their peers.
+- Students will not skip reading or assignments in general.
+- Students will not disrespect peer contributions in work, code, speech, or otherwise.
```
Comment (mariakimheinert) [inline conduct.md]: Could this point be expanded into a couple of points that are more specific about what disrespect looks like?

File: conduct.md (hunk 9 above)
Comment (mariakimheinert) [inline conduct.md]: There was some discussion in issue #1 about the use of contractions, such as "it's". The latest comment indicated that one group was not using contractions. Is this something that a consensus among all teams has been reached about?

**Review by mariakimheinert — CHANGES_REQUESTED** [review]
Hello @ullrichd21, @LiamBlack3, @jackturner83, and @pkelly55! Thanks for all of your hard work on these additions to the Code of Conduct. I've requested some changes below. Please let me know if you have any questions about my comments!

File: conduct.md [hunk 18]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
+Communication is key to both successful and sustainable code. However it's important that all communication between students and team members follows the three pillars of Respect, Trust, and Humility. In general, students will *respect* their peers, *trust* their contributions and thoughts, and practice *humility* in all aspects of communication, in person, on GitHub, in the Discord, and otherwise.
 
 #### Acceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Students should communicate in a professional and respectful manner.
+- Students should use words like "Please" and "Thank you".
+- Students will allow other members and teams speak freely.
```
Comment (solisa986) [inline conduct.md]: there should be a "to" in-between teams and speak.

File: conduct.md (hunk 12 above)
Comment (solisa986) [inline conduct.md]: I agree with @mariakimheinert 's assessment of the last part. Maybe something like that working as a team means "keeping a professional tone" with regard to development work.

File: conduct.md (hunk 11 above)
Comment (solisa986) [inline conduct.md]: there should be "based on" added after "will be" in the first sentence.

Comment (solisa986) [inline conduct.md]: there should be a comma after 'learn' in the last sentence

**Review by solisa986 — CHANGES_REQUESTED** [review]

File: conduct.md (hunk 9 above)
Comment (ullrichd21) [reply conduct.md]: To my knowledge we hadn't even talked about it, so I'll just remove the contraction because I think it's more professional.

File: conduct.md (hunk 12 above)
Comment (ullrichd21) [reply conduct.md]: I didn't write this though I think you make a good point here. I think we should either remove that line or change it to specifically "negative emotions".

File: conduct.md (hunk 17 above)
Comment (ullrichd21) [reply conduct.md]: I think we talked a lot about being respectful so I feel it should be self explanatory? I don't want to expand and make this area too wordy?

File: conduct.md [hunk 19]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
+Participation for students will be based on a multitude of factors. We all have bad days and we all have topics we know more or less about, but how many times a student speaks a day will not solely make up their participation. It is being prepared each day, having done the prerequisite work, coming to class with an open mind to learn, and making an effort to be apart of either small group or large group conversations that will reflect students participation.
 
 #### Acceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Students will respect all peers' opinions.
+- Students will show up to every class ready to learn and engage with the class.
+- Students will evaluate their own participation and the participation of their group members by completing the participation surveys at the end of each week.
```
Comment (noorbuchi) [inline conduct.md:77]: Quick suggestion for this section:

I think it would be nice to have a point on fairly splitting up work on group assignments. This will ensure that the majority of the work is not assigned to one person and  that other members contribute to the group effort.

If this is already addressed or mentioned in a different section, please disregard this comment.

**Review by noorbuchi — DISMISSED** [review]
I think everything looks good. I added a small suggestion for the participation section and I'd like to hear what you think about it. If it's not something you want to add, that's totally fine.

**Review by nathandloria — DISMISSED** [review]
I think that all of these changes look great! Thanks for all of the hard work you guys put into revising the original work to where it is now!

**Review by noorbuchi — DISMISSED** [review]

**Review by PaigeCD — DISMISSED** [review]

File: conduct.md [hunk 20]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
 
 #### Unacceptable Conduct
 
-- TODO
-- TODO
-- TODO
+- Students will not interrupt their peers.
+- Students will not skip reading or assignments.
+- Students will not disrespect peer contributions in work, code, speech, or otherwise. For example, ignoring certain students in your group or belittling the ideas of others.
```
Comment (mariakimheinert) [inline conduct.md]: Could this be reworded to eliminate the sentence fragment, "For example, ignoring certain students in your group or belittling the ideas of others."? Maybe, "Students will not disrespect peer contributions in work, code, speech, or otherwise by, for example, ignoring certain students in their group or belittling the ideas of others."?

File: conduct.md [hunk 21]
```diff
@@ -52,35 +52,35 @@ TODO: Write an introductory paragraph.
...
-- TODO
-- TODO
+- Students should not communicate in a way that offends or disrespects others.
+- Students should not interrupt others--we all have something to share.
+- Students should not show signs of passive aggressive behavior.
 
 ### Participation
 
-TODO: Write an introductory paragraph.
+Participation for students will be based on a multitude of factors. We all have bad days and we all have topics we know more or less about, but how many times a student speaks a day will not solely make up their participation. It is being prepared each day, having done the prerequisite work, coming to class with an open mind to learn, and making an effort to be apart of either small group or large group conversations that will reflect students participation.
```
Comment (mariakimheinert) [inline conduct.md]: "be apart of" should become "be a part of"

**Review by pkelly55 — DISMISSED** [review]
Looks good

**Review by nathandloria — DISMISSED** [review]
💯

**Review by mariakimheinert — CHANGES_REQUESTED** [review]
Team 4, I've left a few suggestions below. Please let me know if you have any questions!

**Review by noorbuchi — DISMISSED** [review]
:shipit:

**Review by PaigeCD — DISMISSED** [review]

**Review by solisa986 — DISMISSED** [review]
Once @mariakimheinert ’s issues are resolved this should be good. Great work guys!

**Review by PaigeCD — APPROVED** [review]

**Review by mariakimheinert — APPROVED** [review]
Looks great, Team 4! Thank you all for your hard work. :shipit:

**Review by noorbuchi — APPROVED** [review]

**Review by nathandloria — APPROVED** [review]

**Review by laurennevill — APPROVED** [review]
Great work!

**Review by lukel43 — APPROVED** [review]
Seems very fitting and fair! Great work on this!

**Review by Jacob-Allebach — APPROVED** [review]
I think these look great! Good descriptors with good examples 👍
